File: jar-client/src/test/java/com/beancounter/client/integ/TestTrnService.java
package com.beancounter.client.integ;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.client.config.ClientConfig;
import com.beancounter.client.services.TrnService;
import com.beancounter.common.contracts.TrnResponse;
import com.beancounter.common.input.TrustedTrnQuery;
import com.beancounter.common.model.Currency;
import com.beancounter.common.model.Portfolio;
import com.beancounter.common.utils.DateUtils;
import com.beancounter.common.utils.PortfolioUtils;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner;
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;

@AutoConfigureStubRunner(
    stubsMode = StubRunnerProperties.StubsMode.LOCAL,
    ids = "org.beancounter:svc-data:+:stubs:10999")
@ImportAutoConfiguration(ClientConfig.class)
@SpringBootTest(classes = ClientConfig.class)
public class TestTrnService {

  private static final Portfolio portfolio;

  static {
    portfolio = PortfolioUtils.getPortfolio(
        "TEST", "NZD Portfolio", new Currency("NZD")
    );
  }

  @Autowired
  private TrnService trnService;

  @Test
  void is_TrnsReturnedForPortfolioId() {
    TrnResponse trnResponse = trnService.query(portfolio);
    assertThat(trnResponse).isNotNull().hasFieldOrProperty("data");
    assertThat(trnResponse.getData()).isNotEmpty();// Don't care about the contents here.
  }

  @Test
  void is_TrnsReturnedForPortfolioAssetId() {
    TrustedTrnQuery query = new TrustedTrnQuery(
        portfolio, new DateUtils().getDate("2020-05-01"), "KMI");
    TrnResponse queryResults = trnService.query(query);
    assertThat(queryResults).isNotNull().hasFieldOrProperty("data");
    assertThat(queryResults.getData()).isNotEmpty();// Don't care about the contents here.
  }


}

File: jar-client/src/test/kotlin/com/beancounter/client/integ/TestTrnService.kt
package com.beancounter.client.integ

import com.beancounter.auth.TokenService
import com.beancounter.client.Constants.Companion.portfolio
import com.beancounter.client.config.ClientConfig
import com.beancounter.client.services.TrnService
import com.beancounter.common.contracts.Payload
import com.beancounter.common.input.TrustedTrnQuery
import com.beancounter.common.utils.DateUtils
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.ImportAutoConfiguration
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties
/**
 * Basic client side transaction tests.
 */
@AutoConfigureStubRunner(stubsMode = StubRunnerProperties.StubsMode.LOCAL, ids = ["org.beancounter:svc-data:+:stubs:10999"])
@ImportAutoConfiguration(ClientConfig::class)
@SpringBootTest(classes = [ClientConfig::class])
class TestTrnService {
    @Autowired
    private lateinit var trnService: TrnService

    @MockBean
    private lateinit var tokenService: TokenService

    @Test
    fun is_TrnsReturnedForPortfolioId() {
        val trnResponse = trnService.query(portfolio)
        assertThat(trnResponse).isNotNull.hasFieldOrProperty(Payload.DATA)
        assertThat(trnResponse.data).isNotEmpty // Don't care about the contents here.
    }

    @Test
    fun is_TrnsReturnedForPortfolioAssetId() {
        val query = TrustedTrnQuery(
            portfolio, DateUtils().getDate("2020-05-01"), "KMI"
        )
        val queryResults = trnService.query(query)
        assertThat(queryResults).isNotNull.hasFieldOrProperty(Payload.DATA)
        assertThat(queryResults.data).isNotEmpty // Don't care about the contents here.
    }
}
